Detectors/CTP/workflow/src/RawDecoderSpec.cxx

```diff
@@ -119,49 +121,134 @@ void RawDecoderSpec::run(framework::ProcessingContext& ctx)
       size_gbt = 0;
       orbit0 = triggerOrbit;
     }
-    // LOG(info) << "payload size:" << payload.size();
+    // Create 128 bit words
+    gsl::span<const uint8_t> payload(it.data(), it.size());
+    gbtword128_t gbtWord128 = 0;
+    int wordCount = 0;
+    std::vector<gbtword128_t> gbtwords128;
+    LOG(info) << "payload size:" << payload.size();
     for (auto payloadWord : payload) {
-      // LOG(info) << wordCount << " payload:" << int(payloadWord);
-      if (wordCount == 15) {
-        wordCount = 0;
-      } else if (wordCount > 9) {
-        wordCount++;
-      } else if (wordCount == 9) {
-        for (int i = 0; i < 8; i++) {
-          gbtWord[wordCount * 8 + i] = bool(int(payloadWord) & (1 << i));
+      int wc = wordCount % 16;
+      // LOG(info) << wordCount << ":" << wc << " payload:" << int(payloadWord);
+      if ((wc == 0) && (wordCount != 0)) {
+        LOG(info) << "128:" << gbtWord128;
+        gbtwords128.push_back(gbtWord128);
+        gbtWord128 = 0;
+      }
+      for (int i = 0; i < 8; i++) {
+        gbtWord128[wc * 8 + i] = bool(int(payloadWord) & (1 << i));
+      }
+      wordCount++;
+    }
+    if (gbtWord128.count()) {
+      gbtwords128.push_back(gbtWord128);
+      LOG(info) << "128:" << gbtWord128;
+    }
+    // Create 80 bits words
+    // mPadding = 0;
+    gbtword80_t remnant = 0;
+    gbtword80_t w80 = 0;
+    std::vector<gbtword80_t> gbtwords80;
+    if (mPadding == 1) {
+      for (auto word : gbtwords128) {
+        LOG(info) << word;
+        std::string str = word.to_string();
+        str = str.substr(48);
+        w80 = std::bitset<80>(str);
+        LOG(info) << "w80:" << w80;
+        gbtwords80.push_back(w80);
+      }
+    } else {
+      int count = 0;
+      for (auto word : gbtwords128) {
+        int countmod = count % 5;
```

Comment:
I don't understand this part, in the case of no-padding, should not you use gbtWord128 as a consecutive chain of 10-byte words (until you hit the alignment filler)?

Reply:
1 ennablePadding - I will follow your example
2. rdh.dataformat - ok.

Reply:
> I don't understand this part, in the case of no-padding, should not you use gbtWord128 as a consecutive chain of 10-byte words (until you hit the alignment filler)?
OK, I may see your point.
My approach is 
1 I create 128 bit words in both cases (padding /no padding)
2 if padding I just take lower 80 bits of 128 bits
3 if not padding I am shifting bits of 128 bit words to cut them to 80 bits appropriately  - I do it brutally as 128*5 = 8*80, i.e. there are only 5 options.

But I think I understand what you wrote above, I think about  it, please, wait.